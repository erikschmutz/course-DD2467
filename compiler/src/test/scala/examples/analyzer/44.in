type 'a tree = Leaf | Node of 'a * 'a tree * 'a tree;;
let t = Node(1, Node(1, Leaf, Leaf), Leaf);;

(*====================
type tree : tree (Leaf | Node of 'a * 'a tree * 'a tree)
val t : Node of int * Node of int * Leaf of () * Leaf of () * Leaf of ()
- :Node of int * Node of int * Leaf of () * Leaf of () * Leaf of ()
====================*)