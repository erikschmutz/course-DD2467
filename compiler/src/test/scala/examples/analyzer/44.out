AnalyzerResult(Constraint(Node,Some(Tuple(List(Integer(),Constraint(Node,Some(Tuple(List(Integer(),Constraint(Leaf,Some(Unit()),List(Integer())),Constraint(Leaf,Some(Unit()),List(Integer()))),List())),List(Integer())),Constraint(Leaf,Some(Unit()),List(Integer()))),List())),List(Integer())),Enviroment(List(TypeEntry(Identifier(tree),Constraints(tree,List(Constraint(Leaf,None,List('a)),Constraint(Node,Some(Tuple(List(Unknown('a),CustomType(tree,List('a)),CustomType(tree,List('a))),List())),List('a))),List('a))),ValEntry(Identifier(t),Constraint(Node,Some(Tuple(List(Integer(),Constraint(Node,Some(Tuple(List(Integer(),Constraint(Leaf,Some(Unit()),List(Integer())),Constraint(Leaf,Some(Unit()),List(Integer()))),List())),List(Integer())),Constraint(Leaf,Some(Unit()),List(Integer()))),List())),List(Integer()))))))